### DISCUSSION TITLE
Navbar with avatar and dropdown menu.

### DISCUSSION THREAD

**Discussion by pekelund** [heroui-inc/heroui#1732] [Help]
Hi, I've just started to try out NextUI, and the first thing I did outside of the template problem failed. :(
(Using next.js 13.4.19)

Perhaps someone have encountered and solved the same problem.

Step 1: Create a new next.js project with:
npx create-next-app -e https://github.com/nextui-org/next-app-template --typescript 

Step 2: Copy code from the documentation about the navbar with avatar and dropdown menu.
Step 3: Simplify the navbar to try to find the issue: see 
[navbar.txt](https://github.com/nextui-org/nextui/files/12785622/navbar.txt)
attachment (rename to navbar.tsx)

running npm run dev.

All works fine until I click the avatar and the dropdown menu is supposed to appear, then I get:

Unhandled Runtime Error
Error: Unknown element <[object Object]> in collection.

Call Stack
$eb2240fc39a57fa5$export$bf788dd355e3a401.getFullNode
../node_modules/@react-stately/collections/dist/import.mjs (205:0)
getFullNode.next
<anonymous>
$eb2240fc39a57fa5$export$bf788dd355e3a401.iterateCollection
../node_modules/@react-stately/collections/dist/import.mjs (161:0)
iterateCollection.next
<anonymous>
[Symbol.iterator]
../node_modules/@react-stately/collections/dist/import.mjs (289:0)
Generator.next
<anonymous>
new $05ca4cd7c4a5a999$export$863faf230ee2118a
../node_modules/@react-stately/tree/dist/import.mjs (79:0)
eval
../node_modules/@react-stately/tree/dist/import.mjs (109:70)
eval
../node_modules/@react-stately/collections/dist/import.mjs (316:0)
mountMemo
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (11771:0)
Object.useMemo
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (12287:0)
useMemo
node_modules/next/dist/compiled/react/cjs/react.development.js (1784:0)
$7613b1592d41b092$export$6cd28814d92fa9c9
../node_modules/@react-stately/collections/dist/import.mjs (310:35)
$875d6693e12af071$export$728d6ba534403756
../node_modules/@react-stately/tree/dist/import.mjs (109:39)
useMenu
../node_modules/@nextui-org/menu/dist/chunk-5CL57MFM.mjs (27:33)
eval
../node_modules/@nextui-org/menu/dist/chunk-PVXU2M5R.mjs (27:13)
renderWithHooks
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (10697:0)
updateForwardRef
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (14682:0)
beginWork$1
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (17373:0)
beginWork
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (25675:0)
performUnitOfWork
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (24526:0)
workLoopSync
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (24242:0)
renderRootSync
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (24197:0)
recoverFromConcurrentError
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (23432:0)
performSyncWorkOnRoot
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (23693:0)
flushSyncWorkAcrossRoots_impl
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (9970:0)
flushSyncWorkOnAllRoots
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (9928:0)
processRootScheduleInMicrotask
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (10072:0)
eval
node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (10243:0)

**Comment by AntonRybakou** [comment]
Same error with me, just trying to make a button with dropdown menu in my navbar:
```
export default function NavMenu(): ReactElement {
  return (
    <Navbar isBordered maxWidth="xl" className="shadow">
      <NavbarBrand>
        <Link href={'/'}>
          <p>Home</p>
        </Link>
      </NavbarBrand>
      <NavbarContent as="div" justify="end">
        <Dropdown>
          <NavbarItem>
            <DropdownTrigger>
              <Button className="transition-transform">Dropdown</Button>
            </DropdownTrigger>
          </NavbarItem>
          <DropdownMenu aria-label="Settings" variant="flat">
            <DropdownItem key={1}>first</DropdownItem>
            <DropdownItem key={2}>second</DropdownItem>
          </DropdownMenu>
        </Dropdown>
      </NavbarContent>
    </Navbar>
  );
}
```
On click on "Dropdown" button getting error:

> 
> Unhandled Runtime Error
> Error: Unknown element <[object Object]> in collection.
> 
> Call Stack
> $eb2240fc39a57fa5$export$bf788dd355e3a401.getFullNode
> node_modules/.pnpm/@react-stately+collections@3.10.3_react@18.2.0/node_modules/@react-stately/collections/dist/import.mjs (205:0)
> getFullNode.next
> <anonymous>
> $eb2240fc39a57fa5$export$bf788dd355e3a401.iterateCollection
> node_modules/.pnpm/@react-stately+collections@3.10.3_react@18.2.0/node_modules/@react-stately/collections/dist/import.mjs (161:0)
> iterateCollection.next
> <anonymous>
> [Symbol.iterator]
> node_modules/.pnpm/@react-stately+collections@3.10.3_react@18.2.0/node_modules/@react-stately/collections/dist/import.mjs (289:0)
> Generator.next
> <anonymous>
> new $05ca4cd7c4a5a999$export$863faf230ee2118a
> node_modules/.pnpm/@react-stately+tree@3.7.4_react@18.2.0/node_modules/@react-stately/tree/dist/import.mjs (78:0)
> eval
> node_modules/.pnpm/@react-stately+tree@3.7.4_react@18.2.0/node_modules/@react-stately/tree/dist/import.mjs (109:70)
> eval
> node_modules/.pnpm/@react-stately+collections@3.10.3_react@18.2.0/node_modules/@react-stately/collections/dist/import.mjs (316:0)
> mountMemo
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (12440:0)
> Object.useMemo
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (13138:0)
> useMemo
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react/cjs/react.development.js (1770:0)
> $7613b1592d41b092$export$6cd28814d92fa9c9
> node_modules/.pnpm/@react-stately+collections@3.10.3_react@18.2.0/node_modules/@react-stately/collections/dist/import.mjs (310:35)
> $875d6693e12af071$export$728d6ba534403756
> node_modules/.pnpm/@react-stately+tree@3.7.4_react@18.2.0/node_modules/@react-stately/tree/dist/import.mjs (109:39)
> useMenu
> node_modules/.pnpm/@nextui-org+menu@2.0.17_@nextui-org+system@2.0.15_@nextui-org+theme@2.1.17_react-dom@18.2.0_r_a4dv63gdbiv5shhsdlaqnauocu/node_modules/@nextui-org/menu/dist/chunk-R4ZQ5Y3Q.mjs (36:33)
> Menu
> node_modules/.pnpm/@nextui-org+menu@2.0.17_@nextui-org+system@2.0.15_@nextui-org+theme@2.1.17_react-dom@18.2.0_r_a4dv63gdbiv5shhsdlaqnauocu/node_modules/@nextui-org/menu/dist/chunk-B24OILC2.mjs (33:13)
> renderWithHooks
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (11009:0)
> updateForwardRef
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (15665:0)
> beginWork$1
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (18408:0)
> beginWork
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (26738:0)
> performUnitOfWork
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (25584:0)
> workLoopSync
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (25300:0)
> renderRootSync
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (25255:0)
> recoverFromConcurrentError
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (24472:0)
> performSyncWorkOnRoot
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (24736:0)
> flushSyncWorkAcrossRoots_impl
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (10274:0)
> flushSyncWorkOnAllRoots
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (10234:0)
> processRootScheduleInMicrotask
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (10379:0)
> eval
> node_modules/.pnpm/next@14.0.2_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js (10550:0)

**Reply by AntonRybakou** [reply]
Directive "use client" in my NavBar component solved an issue for me (NextJS project)

**Reply by slg1309** [reply]
i love you @AntonRybakou !!! spent 2 hours trying to solve it!
